### ISSUE TITLE
Failed pull with pending changes causes loss of changes.

### ISSUE THREAD

**Issue by RonnBlack** [microsoft/vscode#108668]
We have written the needed data into your clipboard because it was too large to send. Please paste.

Issue Type: <b>Bug</b>

A recent enhancement to VSCode was to automatically stash any pending changes changes before a pull and then pop the stash after completion of the stash. If the pull fails you lose all changes if you attempt to retry the pull.

VS Code version: Code 1.50.0 (93c2f0fbf16c5a4b10e4d5f89737d9c2c25488a3, 2020-10-07T06:00:00.397Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|64.00GB (4.69GB free)|
|Process Argv|-psn_0_131104 --crash-reporter-id c557830a-ef09-4cf7-998f-e6d0228604a1|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (40)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|11.3.1
ACF-Snippet|ant|1.9.0
path-intellisense|chr|2.3.0
woocommerce|cla|1.1.0
wpcs-whitelist-flags|cla|1.1.0
bracket-pair-colorizer-2|Coe|0.2.0
apt-source-list-syntax|dei|0.1.4
xml|Dot|2.5.1
gitlens|eam|10.2.2
vscode-npm-script|eg2|0.3.13
vscode-great-icons|emm|2.1.57
json-tools|eri|1.0.2
json-pretty-printer|eus|1.1.0
php-debug|fel|1.13.0
php-intellisense|fel|2.3.14
php-pack|fel|1.0.2
vscode-solution-explorer|fer|0.3.11
terminal|for|0.0.10
terraform|has|2.2.3
wordpress|hri|1.0.1
rest-client|hum|0.24.3
phpcs|ika|1.0.5
easy-icons|jam|0.3.1
Angular2|joh|9.1.2
vscode-docker|ms-|1.6.0
csharp|ms-|1.23.3
remote-wsl|ms-|0.50.0
azure-account|ms-|0.9.3
powershell|ms-|2020.6.0
vs-keybindings|ms-|0.2.0
vscode-typescript-tslint-plugin|ms-|1.2.3
team|ms-|1.161.1
debugger-for-chrome|msj|4.12.11
php-docblocker|nei|2.1.0
wordpress-suite|pet|0.0.6
wordpress-suite-debug-watcher|pet|0.0.6
sort-lines|Tyr|1.9.0
vscode-icons|vsc|11.0.0
wordpress-toolbox|wor|1.3.8
better-align|wwm|1.1.6

(9 theme extensions excluded)

</details>

**Comment by joaomoreno** [comment]
What do you mean by _you lose all changes_? Did you look for the changes in the stash?

**Comment by RonnBlack** [comment]
Specifically the following.
I had Uncommitted changes that were tracked but most likely not staged.
When I attempted the pull it failed. To the best of my recollection the error was something about the account admin requires single sign-on.

I’m not certain about this but to the best of my memory I believe at this point the initially showed but after a brief moment they disappeared. I attributed this to the files having been stashed and the source control display finally catching up with reality. In hindsight it was a mistake to not check for the stash at that time but I assumed it would still be there afterwards if it didn’t recover.

I attempted the pull again which succeeded but the modified files didn’t show up again so I looked for stashes but I only saw stashes that I had created earlier that were named a specific way.

My guess is that it cleaned up or overwrote the earlier stash when it did the new pull.
